rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/TestFileDownloader.java

```diff
@@ -38,52 +37,57 @@
 {
   private static final Log log = LogFactory.getLog(TestFileDownloader.class);
 
-  private static final String testDirectoryPrefix = System.getProperty("java.io.tmpdir") + "/TestFileDownloader/";
-  private static final String testDirectory = testDirectoryPrefix + "dir0";
-  private Configuration conf;
+  private static final String TEST_CACHE_DIR_PREFIX = BookKeeperTestUtils.getTestCacheDirPrefix("TestFileDownloader");
+  private static final int TEST_MAX_DISKS = 1;
+
+  private final Configuration conf = new Configuration();
 
   @BeforeMethod
   public void setUp() throws Exception
   {
-    conf = new Configuration();
-    CacheConfig.setCacheDataDirPrefix(conf, testDirectoryPrefix + "dir");
-    CacheConfig.setMaxDisks(conf, 1);
-    Files.createDirectories(Paths.get(testDirectory, CacheConfig.getCacheDataDirSuffix(conf)));
+    CacheConfig.setCacheDataDirPrefix(conf, TEST_CACHE_DIR_PREFIX);
+
+    BookKeeperTestUtils.createCacheParentDirectories(CacheConfig.getCacheDirPrefixList(conf), TEST_MAX_DISKS);
+    CacheUtil.createCacheDirectories(conf);
   }
 
   @AfterMethod
   public void tearDown() throws Exception
   {
-    Files.walkFileTree(Paths.get(testDirectory), new DeleteFileVisitor());
-    Files.deleteIfExists(Paths.get(testDirectory));
+    BookKeeperTestUtils.removeCacheParentDirectories(CacheConfig.getCacheDirPrefixList(conf), TEST_MAX_DISKS);
```

Comment:
Even though we don't have more than one test method, lets have this in beforeclass and afterclass to make it consistent. Also it will help in future when we will add more test method in this class

Reply:
@abhishekdas99 Makes sense. Moved to `@AfterClass`